packages/modules/order/src/models/credit-line.ts

```diff
@@ -0,0 +1,107 @@
+import { BigNumberRawValue, DAL } from "@medusajs/framework/types"
+import {
+  BigNumber,
+  createPsqlIndexStatementHelper,
+  generateEntityId,
+  MikroOrmBigNumberProperty,
+} from "@medusajs/framework/utils"
+import {
+  BeforeCreate,
+  Entity,
+  ManyToOne,
+  OnInit,
+  OptionalProps,
+  PrimaryKey,
+  Property,
+  Rel,
+} from "@mikro-orm/core"
+import Order from "./order"
+
+type OptionalLineItemProps = DAL.ModelDateColumns
+
+const tableName = "order_credit_line"
+const OrderIdIndex = createPsqlIndexStatementHelper({
+  tableName,
+  columns: ["order_id"],
+  where: "deleted_at IS NOT NULL",
+})
+
+const DeletedAtIndex = createPsqlIndexStatementHelper({
+  tableName,
+  columns: "deleted_at",
+  where: "deleted_at IS NOT NULL",
+})
+
+@Entity({ tableName })
+export default class OrderCreditLine {
+  [OptionalProps]?: OptionalLineItemProps
+
+  @PrimaryKey({ columnType: "text" })
+  id: string
+
+  @ManyToOne({
+    entity: () => Order,
+    mapToPk: true,
+    fieldName: "order_id",
+    columnType: "text",
+  })
+  @OrderIdIndex.MikroORMIndex()
+  order_id: string
+
+  @ManyToOne(() => Order, {
+    persist: false,
+  })
+  order: Rel<Order>
+
+  @Property({
+    columnType: "text",
+    nullable: true,
+  })
+  reference: string | null = null
+
+  @Property({
+    columnType: "text",
+    nullable: true,
+  })
+  reference_id: string | null = null
```

Comment:
question: How will these references be used in practise?

Reply:
Its the resource that will be the enabler of this operation, in this case payment-collection and its id.

We can scope it down to payments or refunds, but that would require creating multiple credit lines, which I can't see a benefit for.